File: placementprepration/tree/swapbst.java
import java.util.*;
class TreeNode{
    int val;
    TreeNode left,right;
    TreeNode(int val){
        this.val=val;
        this.left=null;
        this.right=null;
    }
}
public class swapbst{
    public static void inorder(TreeNode root, List<Integer> vals){
        if(root==null){
            return ;
        }
        inorder(root.left, vals);
        vals.add(root.val);
        inorder(root.right,vals);
    }
    public static void replacewithorted(TreeNode root, List<Integer>vals, int [] index){
        if(root==null){
            return;
        }
        replacewithorted(root.left, vals, index);
        root.val=vals.get(index[0]++);
        replacewithorted(root.right, vals, index);
    }
    public static void recoverbst(TreeNode root){
        List<Integer>vals=new ArrayList<>();
        inorder(root,vals);
        Collections.sort(vals);
        int[] index={0};
        replacewithorted(root, vals, index);
    }
}

File: placementprepration/tree/swapbst.py
import sys


class TreeNode:
    def __init__(self, val):
        self.val = val
        self.left = None
        self.right = None


def inorder(root, vals):
    if not root:
        return
    inorder(root.left, vals)
    vals.append(root.val)
    inorder(root.right, vals)


def replace_with_sorted(root, vals_iter):
    if not root:
        return
    replace_with_sorted(root.left, vals_iter)
    root.val = next(vals_iter)
    replace_with_sorted(root.right, vals_iter)


def recover_bst(root):
    vals = []
    inorder(root, vals)
    vals.sort()
    replace_with_sorted(root, iter(vals))


def main():
    # Placeholder for building a tree and calling recover_bst
    pass


if __name__ == "__main__":
    main()
